mdsobjects/python/tree.py

```diff
@@ -874,6 +876,22 @@ def getVersionDate():
             raise _exc.MDSplusException(status)
         return _scr.Uint64(dt.value).date
 
+    @classmethodX
+    def incrementCurrent(self,tree=None,inc=1):
+        """Increments current shot for specified tree by inc
+        @param tree: Name of tree
+        @type tree: str
+        @param inc: Increment (default: 1)
+        @return shot: New current shot number for the specified tree
+        @rtype int
+        """
+        if isinstance(self,(Tree,)): # instancemethod: args shifted by one
+            if tree is not None: inc = tree
+            tree = self.tree
+        shot = Tree.getCurrent(tree)+inc
```

Comment:
You can't use `Tree.getCurrent(tree)` here because if there is no current shot it returns an exception, you need to instead use `TreeGetCurrentShotId` (which will return 0) or add an exception handler

Reply:
but also if you hav no current you cannot increment it so the exception is justified, no?